### PR TITLE
fix: adjust dex sort rule

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx [hunk 1]
```diff
@@ -443,11 +443,24 @@ export function Trader(props: TraderProps) {
...
+                .sort(
+                    (
+                        { finalPrice: finalPriceA, gas: { value: gasA }, value: valueA },
+                        { finalPrice: finalPriceB, gas: { value: gasB }, value: valueB },
+                    ) => {
+                        let a = finalPriceA
+                        let b = finalPriceB
+
+                        // if the estimated gas isn't exist
+                        if (!gasA && !gasB) {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx]: Lost conditions for if one of a pair is missing gas field.

**Comment by wenluomask** [comment]
1. There will be an odd estimate received in some case;
Screen Shot 2022-01-11 at 6 27 25 PM

2. The order of best price seems has problem;
Screen Shot 2022-01-11 at 6 41 16 PM

**Comment by wenluomask** [comment]
Checked.

**Comment by wenluomask** [comment]
Checked.

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/hooks/useGasConfig.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        try {
+            const network = getNetworkTypeFromChainId(chainId)
+            if (gasConfig) {
+                return new BigNumber(
+                    (network === NetworkType.Ethereum ? gasConfig.maxFeePerGas : gasConfig.gasPrice) ?? 0,
+                ).toString()
+            } else {
+                if (network === NetworkType.Ethereum) {
+                    const response = await WalletRPC.getEstimateGasFees(chainId)
+                    const maxFeePerGas = formatGweiToWei(response?.medium?.suggestedMaxFeePerGas ?? 0).toString()
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trader/hooks/useGasConfig.ts]:
```suggestion
                    const maxFeePerGas = formatGweiToWei(response?.medium?.suggestedMaxFeePerGas ?? 0).toFixed()
```

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/hooks/useGasConfig.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                    const maxPriorityFeePerGas = formatGweiToWei(
+                        response?.medium?.suggestedMaxPriorityFeePerGas ?? 0,
+                    ).toString()
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trader/hooks/useGasConfig.ts]:
```suggestion
                    ).toFixed()
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**